tensordict/memmap.py

```diff
@@ -254,6 +254,37 @@ def from_tensor(
         out.copy_(tensor)
         return out
 
+    @classmethod
+    def empty_like(
+        cls,
+        tensor: Union[torch.Tensor, MemmapTensor, np.ndarray],
+        transfer_ownership=False,
+        prefix=None,
+        filename: Optional[str] = None,
+        mode: str = "r+",
+    ) -> MemmapTensor:
+        if isinstance(tensor, np.ndarray):
+            raise TypeError(
+                "Convert input to torch.Tensor before calling MemmapTensor."
+            )
+        device = tensor.device if hasattr(tensor, "device") else torch.device("cpu")
+        dtype = (
+            tensor.dtype
+            if isinstance(tensor, (torch.Tensor, MemmapTensor))
+            else numpy_to_torch_dtype_dict[tensor.dtype.name]
+        )
```

Comment:
If `tensor` is an instance of `np.ndarray` then this code isn't actually reachable (we would have raised a `TypeError` on line 267) so I think the ternary operator isn't necessary.